File: solutions/go/word-count/1/word_count.go
// Package wordcount deals with words and numbers.
package wordcount

import (
	"regexp"
	"strings"
)

type Frequency map[string]int

// Frequency counts occurrences of words within a string.
func WordCount(s string) Frequency {
	counts := make(map[string]int)
	re := regexp.MustCompile(`[ ,\r\n]+`)
	words := re.Split(s, -1)
	unclean := regexp.MustCompile(`[^a-zA-Z0-9']+`)
	for _, word := range words {
		if word == "" {
			continue
		}
		cleanWord := unclean.ReplaceAllString(word, "")
		cleanWord = strings.Trim(cleanWord, `'`)
		counts[strings.ToLower(cleanWord)] += 1
	}
	return counts
}

File: solutions/julia/word-count/1/word-count.jl
function wordcount(sentence::AbstractString)
    alphanum = r"[a-zA-Z0-9]"
    notalphanum = r"[^a-zA-Z0-9']"
    if !occursin(alphanum, sentence)
        return Dict()
    end
    d = Dict()
    words = split(sentence, r"[ ,\r\n]+")
    for word in words
        word = replace(word, notalphanum => "")
        word = lowercase(word)
        word = strip(word, ['''])
        d[word] = get(d, word, 0) + 1
    end
    d
end
